Modules/@babylonjs/react-native-basekit/scripts/install.js

```diff
@@ -0,0 +1,174 @@
+      process.exit(1);
+    }
+  })
+  .on('error', (err) => {
+    console.error('Error downloading the file:', err);
+    process.exit(1);
+  });
+};
+
+function Install() {
+  // get Babylon React Native package version
+  const packageJsonPath = process.env.npm_package_json;
+  const packageJson = require(packageJsonPath);
+  if (!packageJson.version) {
+    console.err("Babylon React Native version not found");
+    process.exit(1);
+  }
+  // get user project react-native version
+  const projectPackageJsonPath = process.env.npm_config_local_prefix + '/package.json';
+
+  // Read the project's package.json file
+  const projectPackageJson = require(projectPackageJsonPath);
+  // Check if the project's package.json has a 'react-native' dependency
+  if (projectPackageJson.dependencies && projectPackageJson.dependencies['react-native']) {
+    const reactNativeVersion = projectPackageJson.dependencies['react-native'];
+    console.log(`Detected React Native version: ${reactNativeVersion}`);
+    reactNativePostfix = "";
+    if (reactNativeVersion >= '0.71') {
+      reactNativePostfix = "0.71";
+    } else if (reactNativeVersion >= '0.70') {
+      reactNativePostfix = "0.70";
+    } else if (reactNativeVersion >= '0.69') {
+      reactNativePostfix = "0.69";
+    } else {
+      console.error("Unsupported react native version.");
+      process.exit(1);
+    }
+    console.log(`Using Babylon React Native version ${reactNativePostfix} from Package version ${packageJson.version}.`);
+
+    const jsArchive = 'react-native.tar.gz';
+    const iosAndroidArchive = `iOSAndroid${reactNativePostfix}.tar.gz`;
+    const windowsArchive = `Windows${reactNativePostfix}.tar.gz`;
+
+    const jsArchiveCachedPath = getCached(packageJson, jsArchive);
+    const iosAndroidArchiveCachedPath = getCached(packageJson, iosAndroidArchive);
+    const windowsArchiveCachedPath = getCached(packageJson, windowsArchive);
+
+    if (!useCachedFile(jsArchiveCachedPath)) {
+      const reactNative = getBinaryUrl(packageJson, jsArchive);
+      downloadExtractAndCache(reactNative, jsArchiveCachedPath);
+    }
+
+    if (!useCachedFile(iosAndroidArchiveCachedPath)) {
+      const reactNativeiOSAndroid = getBinaryUrl(packageJson, iosAndroidArchive);
+      downloadExtractAndCache(reactNativeiOSAndroid, iosAndroidArchiveCachedPath);
+    }
+
+    // check and download Windows binary if react-native-windows is found in project package
+    const reactNativeWindowsVersion = projectPackageJson.dependencies['react-native-windows'];
+    if (reactNativeWindowsVersion) {
+      console.log("react-native-windows detected.");
+      if (!useCachedFile(windowsArchiveCachedPath)) {
+        const reactNativeiOSWindows = getBinaryUrl(packageJson, windowsArchive);
+        downloadExtractAndCache(reactNativeiOSWindows, windowsArchiveCachedPath);
+      }
+    }
+  } else {
+    console.error("No react-native version found for BabylonReactNative.");
+    process.exit(1);
```

Comment:
general idea (totally not a must) if you want to catch the exact error (for example during CI testing), give them different code values